api/app/Models/User.php

```diff
@@ -472,6 +472,49 @@ public function filterByEquity(Builder $query, array $equity): Builder
         return $query;
     }
 
+    public function filterByGeneralSearch(Builder $query, ?string $search): Builder
+    {
+        if ($search) {
+            $query->where(function($query) use ($search) {
+                $query->where('first_name', "like", "%{$search}%")
+                      ->orWhere('last_name', "like", "%{$search}%")
+                      ->orWhere('email', "like", "%{$search}%")
+                      ->orWhere('telephone', "like", "%{$search}%");
+            });
+        }
+        return $query;
+    }
+
+    public function filterByName(Builder $query, ?string $name): Builder
+    {
+        if ($name) {
+            $splitName = explode(" ", $name);
+            $query->where(function($query) use ($splitName) {
+                foreach($splitName as $index => $value){
+                    $query->where('first_name', "like", "%{$value}%")
+                        ->orWhere('last_name', "like", "%{$value}%");
+                }
+            });
```

Comment:
I think this is good, in terms of OR groupings? Looks to me like this will require that each segment of splitName match at least one firstName or lastName, which makes sense to me. Of course, you'll want some good tests for this.